src/main/java/com/mic/betterslimes/entity/slimes/KingSlime.java

```diff
@@ -26,128 +32,389 @@
-	}
-
-	public void removeTrackingPlayer(EntityPlayerMP player) {
-		super.removeTrackingPlayer(player);
-		this.bossInfo.removePlayer(player);
-	}
-
-	@Override
-	protected EntityBetterSlime createInstance() {
-		return new BlueSlime(this.world);
-	}
-
-	@Override
-	public boolean getCanSpawnHere() {
-
-		if (this.world.getWorldInfo().getTerrainType().handleSlimeSpawnReduction(rand, world)) {
-			return false;
-		} else {
-			if (this.world.getDifficulty() != EnumDifficulty.PEACEFUL) {
-
-				return true;
-
-			}
-
-			return false;
-		}
-	}
-
-	@Override
-	protected int getAttackStrength() {
-		return (int) (super.getAttackStrength() * attackMod);
-	}
-
-	@Override
-	protected void applyEntityAttributes() {
-		super.applyEntityAttributes();
-		this.getEntityAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(40.0D);
-	}
-
-	@Nullable
-	protected ResourceLocation getLootTable() {
-		return BetterSlimes.kingSlimeLT;
-	}
+                KnightSlime b;
+                for (int x = 0; x < 4; x++) {
+                    b = new KnightSlime(this.world);
+                    b.setSlimeSize(2, true);
+                    b.setLocationAndAngles(this.posX + rand.nextInt(10) - 5, this.posY + rand.nextInt(1) + 1,
+                            this.posZ + rand.nextInt(10) - 5, this.rotationYaw, this.rotationPitch);
+                    this.world.spawnEntity(b);
+                }
+            }
+
+            this.setSpawnTime(j1);
+        } else {
+            this.setSpawnTime(240);
+        }
+        super.updateAITasks();
+        this.bossInfo.setPercent(this.getHealth() / this.getMaxHealth());
+    }
+
+    @Override
+    public void addTrackingPlayer(EntityPlayerMP player) {
+        super.addTrackingPlayer(player);
+        this.bossInfo.addPlayer(player);
+
+    }
+
+    public void setCustomNameTag(String name) {
+        super.setCustomNameTag(name);
+        this.bossInfo.setName(this.getDisplayName());
+    }
+
+    public void writeEntityToNBT(NBTTagCompound compound) {
+        super.writeEntityToNBT(compound);
+        if (configLoaded) {
+            compound.setInteger("Spawn", this.getSpawnTime());
+            compound.setInteger("leapCooldown", leapCooldown);
+            compound.setInteger("leapWarning", leapWarning);
+            compound.setFloat("leapVelocityMultiplierY", leapVelocityMultiplierY);
+            compound.setFloat("leapVelocityMultiplierXZ", leapVelocityMultiplierXZ);
+            compound.setFloat("explodeDamage", explodeDamage);
+            compound.setInteger("explodeRange", explodeRange);
+            compound.setBoolean("spawnMinions", spawnMinions);
+            nbtSet = true;
+        }
+    }
+
+    public void readEntityFromNBT(NBTTagCompound compound) {
+        super.readEntityFromNBT(compound);
+        this.setSpawnTime(compound.getInteger("Spawn"));
+        if (this.hasCustomName()) {
+            this.bossInfo.setName(this.getDisplayName());
+        }
+        if (configLoaded  && nbtSet) {
```

Comment:
I'm pretty sure nbtSet has no use because it's not set if the entity is spawned via commands. The way to prevent null values is to check if the nbt compound contains the key before setting the variable to the key's value.